Kafka configuration parsing fails with `java.lang.NumberFormatException: For input string: "org.apache.kafka.common.serialization.StringSerializer"`

In the native mode, a Quarkus app that uses reactive-messaging-kafka fails with the error below.

It seems the problem is in the fact that SmallRye Config wraps the error in parsing value in an IllegalArgumentException.

```
2019-05-15 16:20:30,472 ERROR [io.sma.rea.mes.imp.LegacyConfiguredStreamFactory] (main) Unable to create the publisher or subscriber during initialization: java.lang.IllegalArgumentException: java.lang.reflect.InvocationTargetException
        at io.smallrye.config.ImplicitConverters$StaticMethodConverter.convert(ImplicitConverters.java:99)
        at io.smallrye.config.SmallRyeConfig.convert(SmallRyeConfig.java:133)
        at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:76)
        at io.smallrye.reactive.messaging.impl.ConnectorConfig.getValue(ConnectorConfig.java:57)
        at io.smallrye.reactive.messaging.spi.ConfigurationHelper.asJsonObject(ConfigurationHelper.java:51)
        at io.smallrye.reactive.messaging.kafka.KafkaSink.<init>(KafkaSink.java:30)
        at io.smallrye.reactive.messaging.kafka.KafkaMessagingProvider.getSubscriberBuilder(KafkaMessagingProvider.java:47)
        at io.smallrye.reactive.messaging.kafka.KafkaMessagingProvider_ClientProxy.getSubscriberBuilder(Unknown Source)
        at io.smallrye.reactive.messaging.impl.LegacyConfiguredStreamFactory.createSubscriberBuilder(LegacyConfiguredStreamFactory.java:134)
        at io.smallrye.reactive.messaging.impl.LegacyConfiguredStreamFactory.lambda$initialize$3(LegacyConfiguredStreamFactory.java:101)
        at java.util.HashMap.forEach(HashMap.java:1289)
        at io.smallrye.reactive.messaging.impl.LegacyConfiguredStreamFactory.initialize(LegacyConfiguredStreamFactory.java:101)
        at io.smallrye.reactive.messaging.impl.LegacyConfiguredStreamFactory_ClientProxy.initialize(Unknown Source)
        at java.util.Iterator.forEachRemaining(Iterator.java:116)
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
        at io.smallrye.reactive.messaging.extension.MediatorManager.initializeAndRun(MediatorManager.java:103)
        at io.smallrye.reactive.messaging.extension.MediatorManager_ClientProxy.initializeAndRun(Unknown Source)
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:18)
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle_Observer_onApplicationStart_4e8937813d9e8faff65c3c07f88fa96615b70e70.notify(Unknown Source)
        at io.quarkus.arc.EventImpl$Notifier.notify(EventImpl.java:244)
        at io.quarkus.arc.EventImpl.fire(EventImpl.java:85)
        at io.quarkus.arc.runtime.LifecycleEventRunner.fireStartupEvent(LifecycleEventRunner.java:39)
        at io.quarkus.arc.runtime.ArcDeploymentTemplate.handleLifecycleEvents(ArcDeploymentTemplate.java:115)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent20.deploy(Unknown Source)
        at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.Application.run(Application.java:213)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Method.java:498)
        at io.smallrye.config.ImplicitConverters$StaticMethodConverter.convert(ImplicitConverters.java:97)
        ... 28 more
Caused by: java.lang.NumberFormatException: For input string: "org.apache.kafka.common.serialization.StringSerializer"
        at java.lang.Integer.parseInt(Integer.java:580)
        at java.lang.Integer.valueOf(Integer.java:766)
        ... 30 more
```

--- a/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/spi/ConfigurationHelper.java
+++ b/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/spi/ConfigurationHelper.java
@@ -51,7 +51,7 @@ public class ConfigurationHelper {
         int i = config.getValue(key, Integer.class);
         json.put(key, i);
         continue;
-      } catch (ClassCastException | NumberFormatException e) {
+      } catch (ClassCastException | IllegalArgumentException e) {
         // Ignore me
       }
 
@@ -59,7 +59,7 @@ public class ConfigurationHelper {
         double d = config.getValue(key, Double.class);
         json.put(key, d);
         continue;
-      } catch (ClassCastException | NumberFormatException e) {
+      } catch (ClassCastException | IllegalArgumentException e) {
         // Ignore me
       }
 

